coderd/workspacebuilds.go

```diff
@@ -383,6 +384,13 @@ func (api *API) postWorkspaceBuilds(rw http.ResponseWriter, r *http.Request) {
 			builder = builder.State(createBuild.ProvisionerState)
 		}
 
+		// Only defer to dynamic parameters if the experiment is enabled.
+		if api.Experiments.Enabled(codersdk.ExperimentDynamicParameters) &&
+			createBuild.EnableDynamicParameters != nil {
+			// Explicit opt-in
+			builder = builder.DynamicParameters(*createBuild.EnableDynamicParameters)
+		}
```

Comment:
I provide the explicit opt in as a way to force through transitions such as `DELETE` if there an issue. It would be unfortunate to have to change a template wide setting to fix a single workspace if we get into a strange state.

Reply:
Can we drop a WARN or something if the build requests dynamic parameters but the experiment is not enabled?